File: peekscan-app/peekscan-app/modules/peekscan-vision/android/PeekScanVisionModule.kt
package expo.modules.peekscanvision

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PeekScanVisionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PeekScanVision")

    // The logic is mainly in C++ called via Frame Processor
    // This module provides a simple sync/async test method
    Function("scanImage") { base64: String ->
      // Bridge to C++ would go here
      mapOf(
        "success" to true,
        "codes" to listOf<Map<String, String>>()
      )
    }
  }
}

File: peekscan-app/peekscan-app/modules/peekscan-vision/ios/PeekScanVisionModule.swift
import ExpoModulesCore

public class PeekScanVisionModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PeekScanVision")

    Function("scanImage") { (base64: String) -> [String: Any] in
      // Bridge to C++ would go here
      return [
        "success": true,
        "codes": []
      ]
    }
  }
}
